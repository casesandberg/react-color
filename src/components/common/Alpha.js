'use strict' /* @flow */

import React from 'react'
import ReactCSS from 'reactcss'
import shallowCompare from 'react-addons-shallow-compare'

import Checkboard from './Checkboard'

export class Alpha extends ReactCSS.Component {
  shouldComponentUpdate = shallowCompare.bind(this, this, arguments[0], arguments[1])

  classes(): any {
    return {
      'default': {
        alpha: {
          Absolute: '0px 0px 0px 0px',
          borderRadius: this.props.radius,
        },
        checkboard: {
          Absolute: '0px 0px 0px 0px',
          overflow: 'hidden',
        },
        gradient: {
          Absolute: '0px 0px 0px 0px',
          background: 'linear-gradient(to right, rgba(' + this.props.rgb.r + ', ' + this.props.rgb.g + ', ' + this.props.rgb.b + ', 0) 0%, rgba(' + this.props.rgb.r + ', ' + this.props.rgb.g + ', ' + this.props.rgb.b + ', 1) 100%)',
          boxShadow: this.props.shadow,
          borderRadius: this.props.radius,
        },
        container: {
          position: 'relative',
          zIndex: '2',
          height: '100%',
          margin: '0px 3px',
        },
        pointer: {
          zIndex: '2',
          position: 'absolute',
          left: this.props.rgb.a * 100 + '%',
        },
        slider: {
          width: '4px',
          borderRadius: '1px',
          height: '8px',
          boxShadow: '0px 0px 2px rgba(0, 0, 0, .6)',
          background: '#fff',
          marginTop: '1px',
          transform: 'translateX(-2px)',
        },
      },
    }
  }

  componentWillUnmount() {
    this.unbindEventListeners()
  }

  handleChange = (e: any, skip: boolean) => {
    !skip && e.preventDefault()
    var container = this.refs.container
    var containerWidth = container.clientWidth
    var left = (e.pageX || e.touches[0].pageX) - (container.getBoundingClientRect().left + window.pageXOffset)

    var a
    if (left < 0) {
      a = 0
    } else if (left > containerWidth) {
      a = 1
    } else {
      a = Math.round(left * 100 / containerWidth) / 100
    }

    if (this.props.a !== a) {
      this.props.onChange({
        h: this.props.hsl.h,
        s: this.props.hsl.s,
        l: this.props.hsl.l,
        a: a,
        source: 'rgb',
      })
    }
  }

  handleMouseDown = (e: any) => {
    this.handleChange(e, true)
    window.addEventListener('mousemove', this.handleChange)
    window.addEventListener('mouseup', this.handleMouseUp)
  }

  handleMouseUp = () => {
    this.unbindEventListeners()
  }

  unbindEventListeners = () => {
    window.removeEventListener('mousemove', this.handleChange)
    window.removeEventListener('mouseup', this.handleMouseUp)
  }

  render(): any {
    var pointer = <div is="slider" />

    if (this.props.pointer) {
      pointer = <this.props.pointer {...this.props} />
    }

    return (
      <div is="alpha">
        <div is="checkboard">
          <Checkboard />
        </div>
        <div is="gradient" />
        <div is="container" ref="container" onMouseDown={ this.handleMouseDown }
            onTouchMove={ this.handleChange }
            onTouchStart={ this.handleChange }>
          <div is="pointer" ref="pointer">
            { pointer }
          </div>
        </div>
      </div>
    )
  }
}

export default Alpha
