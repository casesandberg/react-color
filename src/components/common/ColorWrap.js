'use strict' /* @flow */

import React from 'react'
import merge from 'merge'
import isPlainObject from 'lodash/isplainobject'
import debounce from 'lodash/debounce'
import color from '../../helpers/color'
import shallowCompare from 'react-addons-shallow-compare'

export const ColorWrap = (Picker) => {
  class ColorPicker extends React.Component {
    shouldComponentUpdate = shallowCompare.bind(this, this, arguments[0], arguments[1]);

    constructor(props: any) {
      super()

      this.state = merge(color.toState(props.color, 0), {
        visible: props.display,
      })

      this.debounce = debounce(function (fn: any, data: any) {
        fn(data)
      }, 100)
    }

    handleChange = (data: any) => {
      data = color.simpleCheckForValidColor(data)
      if (data) {
        var colors = color.toState(data, data.h || this.state.oldHue)
        this.setState(colors)
        this.props.onChangeComplete && this.debounce(this.props.onChangeComplete, colors)
        this.props.onChange && this.props.onChange(colors)
      }
    }

    componentWillReceiveProps(nextProps: any) {
      this.setState(merge(color.toState(nextProps.color, this.state.oldHue), {
        visible: nextProps.display,
      }))
    }

    render(): any {
      return <Picker {...this.props} {...this.state} onChange={ this.handleChange } />
    }
  }

  ColorPicker.defaultProps = {
    color: {
      h: 250,
      s: .50,
      l: .20,
      a: 1,
    },
  }

  return ColorPicker
}

export default ColorWrap
