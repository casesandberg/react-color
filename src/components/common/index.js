export { default as Alpha } from './Alpha'
export { default as Checkboard } from './Checkboard'
export { default as EditableInput } from './EditableInput'
export { default as Hue } from './Hue'
export { default as Raised } from './Raised'
export { default as Saturation } from './Saturation'
export { default as ColorWrap } from './ColorWrap'
export { default as Swatch } from './Swatch'
