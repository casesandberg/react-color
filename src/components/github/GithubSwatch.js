import React from 'react'
import reactCSS, { handleHover } from 'reactcss'

import { Swatch } from '../common'

export const GithubSwatch = ({ hover, color, onClick, onSwatchHover }) => {
  const hoverSwatch = {
    position: 'relative',
    zIndex: '2',
    outline: '2px solid #fff',
    boxShadow: '0 0 5px 2px rgba(0,0,0,0.25)',
  }

  const styles = reactCSS({
    'default': {
      swatch: {
        width: '25px',
        height: '25px',
        fontSize: '0',
      },
    },
    'hover': {
      swatch: hoverSwatch,
    },
  }, { hover })

  return (
    <div style={ styles.swatch }>
      <Swatch
        color={ color }
        onClick={ onClick }
        onHover={ onSwatchHover }
        focusStyle={ hoverSwatch }
      />
    </div>
  )
}

export default handleHover(GithubSwatch)
