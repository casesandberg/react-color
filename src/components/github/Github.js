import React from 'react'
import PropTypes from 'prop-types'
import reactCSS from 'reactcss'
import map from 'lodash/map'
import merge from 'lodash/merge'

import { ColorWrap } from '../common'
import GithubSwatch from './GithubSwatch'

export const Github = ({ width, colors, onChange, onSwatchHover, triangle,
  styles: passedStyles = {}, className = '' }) => {
  const styles = reactCSS(merge({
    'default': {
      card: {
        width,
        background: '#fff',
        border: '1px solid rgba(0,0,0,0.2)',
        boxShadow: '0 3px 12px rgba(0,0,0,0.15)',
        borderRadius: '4px',
        position: 'relative',
        padding: '5px',
        display: 'flex',
        flexWrap: 'wrap',
      },
      triangle: {
        position: 'absolute',
        border: '7px solid transparent',
        borderBottomColor: '#fff',
      },
      triangleShadow: {
        position: 'absolute',
        border: '8px solid transparent',
        borderBottomColor: 'rgba(0,0,0,0.15)',
      },
    },
    'hide-triangle': {
      triangle: {
        display: 'none',
      },
      triangleShadow: {
        display: 'none',
      },
    },
    'top-left-triangle': {
      triangle: {
        top: '-14px',
        left: '10px',
      },
      triangleShadow: {
        top: '-16px',
        left: '9px',
      },
    },
    'top-right-triangle': {
      triangle: {
        top: '-14px',
        right: '10px',
      },
      triangleShadow: {
        top: '-16px',
        right: '9px',
      },
    },
    'bottom-left-triangle': {
      triangle: {
        top: '35px',
        left: '10px',
        transform: 'rotate(180deg)',
      },
      triangleShadow: {
        top: '37px',
        left: '9px',
        transform: 'rotate(180deg)',
      },
    },
    'bottom-right-triangle': {
      triangle: {
        top: '35px',
        right: '10px',
        transform: 'rotate(180deg)',
      },
      triangleShadow: {
        top: '37px',
        right: '9px',
        transform: 'rotate(180deg)',
      },
    },
  }, passedStyles), {
    'hide-triangle': triangle === 'hide',
    'top-left-triangle': triangle === 'top-left',
    'top-right-triangle': triangle === 'top-right',
    'bottom-left-triangle': triangle === 'bottom-left',
    'bottom-right-triangle': triangle === 'bottom-right',
  })

  const handleChange = (hex, e) => onChange({ hex, source: 'hex' }, e)

  return (
    <div style={ styles.card } className={ `github-picker ${ className }` }>
      <div style={ styles.triangleShadow } />
      <div style={ styles.triangle } />
      { map(colors, c => (
        <GithubSwatch
          color={ c }
          key={ c }
          onClick={ handleChange }
          onSwatchHover={ onSwatchHover }
        />
      )) }
    </div>
  )
}

Github.propTypes = {
  width: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  colors: PropTypes.arrayOf(PropTypes.string),
  triangle: PropTypes.oneOf(['hide', 'top-left', 'top-right', 'bottom-left', 'bottom-right']),
  styles: PropTypes.object,
}

Github.defaultProps = {
  width: 200,
  colors: ['#B80000', '#DB3E00', '#FCCB00', '#008B02', '#006B76', '#1273DE', '#004DCF', '#5300EB',
    '#EB9694', '#FAD0C3', '#FEF3BD', '#C1E1C5', '#BEDADC', '#C4DEF6', '#BED3F3', '#D4C4FB'],
  triangle: 'top-left',
  styles: {},
}

export default ColorWrap(Github)
