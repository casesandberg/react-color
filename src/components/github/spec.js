/* global test, expect */

import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme';
import { red } from '../../helpers/color'

import Github from './Github'
import GithubSwatch from './GithubSwatch'
import { Swatch } from '../common'

test('Github renders correctly', () => {
  const tree = renderer.create(
    <Github { ...red } />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('Github with onSwatchHover events correctly', () => {
  const hoverSpy = jest.fn()
  const tree = mount(
    <Github onSwatchHover={hoverSpy} />
  )
  expect(hoverSpy).toHaveBeenCalledTimes(0)
  const swatches = tree.find(Swatch);
  swatches.at(0).childAt(0).simulate('mouseOver')

  expect(hoverSpy).toHaveBeenCalled()
})

test('Github `triangle="none"`', () => {
  const tree = renderer.create(
    <Github { ...red } triangle="none" />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('Github `triangle="top-right"`', () => {
  const tree = renderer.create(
    <Github { ...red } triangle="top-right" />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('GithubSwatch renders correctly', () => {
  const tree = renderer.create(
    <GithubSwatch color="#333" />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
