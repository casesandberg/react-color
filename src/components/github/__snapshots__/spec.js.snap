exports[`test Github \`triangle="none"\` 1`] = `
<div
  className="github-picker"
  style={
    Object {
      "MozBorderRadius": "4px",
      "MozBoxShadow": "0 3px 12px rgba(0,0,0,0.15)",
      "OBorderRadius": "4px",
      "OBoxShadow": "0 3px 12px rgba(0,0,0,0.15)",
      "WebkitBorderRadius": "4px",
      "WebkitBoxShadow": "0 3px 12px rgba(0,0,0,0.15)",
      "background": "#fff",
      "border": "1px solid rgba(0,0,0,0.2)",
      "borderRadius": "4px",
      "boxShadow": "0 3px 12px rgba(0,0,0,0.15)",
      "display": "flex",
      "flexWrap": "wrap",
      "msBorderRadius": "4px",
      "msBoxShadow": "0 3px 12px rgba(0,0,0,0.15)",
      "padding": "5px",
      "position": "relative",
      "width": "200px",
    }
  }>
  <div
    style={
      Object {
        "border": "8px solid transparent",
        "borderBottomColor": "rgba(0,0,0,0.15)",
        "position": "absolute",
      }
    } />
  <div
    style={
      Object {
        "border": "7px solid transparent",
        "borderBottomColor": "#fff",
        "position": "absolute",
      }
    } />
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "height": "25px",
          "width": "25px",
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "background": "#B80000",
            "cursor": "pointer",
            "height": "100%",
            "width": "100%",
          }
        }
        title="#B80000" />
    </div>
  </span>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "height": "25px",
          "width": "25px",
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "background": "#DB3E00",
            "cursor": "pointer",
            "height": "100%",
            "width": "100%",
          }
        }
        title="#DB3E00" />
    </div>
  </span>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "height": "25px",
          "width": "25px",
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "background": "#FCCB00",
            "cursor": "pointer",
            "height": "100%",
            "width": "100%",
          }
        }
        title="#FCCB00" />
    </div>
  </span>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "height": "25px",
          "width": "25px",
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "background": "#008B02",
            "cursor": "pointer",
            "height": "100%",
            "width": "100%",
          }
        }
        title="#008B02" />
    </div>
  </span>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "height": "25px",
          "width": "25px",
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "background": "#006B76",
            "cursor": "pointer",
            "height": "100%",
            "width": "100%",
          }
        }
        title="#006B76" />
    </div>
  </span>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "height": "25px",
          "width": "25px",
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "background": "#1273DE",
            "cursor": "pointer",
            "height": "100%",
            "width": "100%",
          }
        }
        title="#1273DE" />
    </div>
  </span>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "height": "25px",
          "width": "25px",
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "background": "#004DCF",
            "cursor": "pointer",
            "height": "100%",
            "width": "100%",
          }
        }
        title="#004DCF" />
    </div>
  </span>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "height": "25px",
          "width": "25px",
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "background": "#5300EB",
            "cursor": "pointer",
            "height": "100%",
            "width": "100%",
          }
        }
        title="#5300EB" />
    </div>
  </span>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "height": "25px",
          "width": "25px",
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "background": "#EB9694",
            "cursor": "pointer",
            "height": "100%",
            "width": "100%",
          }
        }
        title="#EB9694" />
    </div>
  </span>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "height": "25px",
          "width": "25px",
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "background": "#FAD0C3",
            "cursor": "pointer",
            "height": "100%",
            "width": "100%",
          }
        }
        title="#FAD0C3" />
    </div>
  </span>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "height": "25px",
          "width": "25px",
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "background": "#FEF3BD",
            "cursor": "pointer",
            "height": "100%",
            "width": "100%",
          }
        }
        title="#FEF3BD" />
    </div>
  </span>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "height": "25px",
          "width": "25px",
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "background": "#C1E1C5",
            "cursor": "pointer",
            "height": "100%",
            "width": "100%",
          }
        }
        title="#C1E1C5" />
    </div>
  </span>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "height": "25px",
          "width": "25px",
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "background": "#BEDADC",
            "cursor": "pointer",
            "height": "100%",
            "width": "100%",
          }
        }
        title="#BEDADC" />
    </div>
  </span>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "height": "25px",
          "width": "25px",
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "background": "#C4DEF6",
            "cursor": "pointer",
            "height": "100%",
            "width": "100%",
          }
        }
        title="#C4DEF6" />
    </div>
  </span>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "height": "25px",
          "width": "25px",
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "background": "#BED3F3",
            "cursor": "pointer",
            "height": "100%",
            "width": "100%",
          }
        }
        title="#BED3F3" />
    </div>
  </span>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "height": "25px",
          "width": "25px",
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "background": "#D4C4FB",
            "cursor": "pointer",
            "height": "100%",
            "width": "100%",
          }
        }
        title="#D4C4FB" />
    </div>
  </span>
</div>
`;

exports[`test Github \`triangle="top-right"\` 1`] = `
<div
  className="github-picker"
  style={
    Object {
      "MozBorderRadius": "4px",
      "MozBoxShadow": "0 3px 12px rgba(0,0,0,0.15)",
      "OBorderRadius": "4px",
      "OBoxShadow": "0 3px 12px rgba(0,0,0,0.15)",
      "WebkitBorderRadius": "4px",
      "WebkitBoxShadow": "0 3px 12px rgba(0,0,0,0.15)",
      "background": "#fff",
      "border": "1px solid rgba(0,0,0,0.2)",
      "borderRadius": "4px",
      "boxShadow": "0 3px 12px rgba(0,0,0,0.15)",
      "display": "flex",
      "flexWrap": "wrap",
      "msBorderRadius": "4px",
      "msBoxShadow": "0 3px 12px rgba(0,0,0,0.15)",
      "padding": "5px",
      "position": "relative",
      "width": "200px",
    }
  }>
  <div
    style={
      Object {
        "border": "8px solid transparent",
        "borderBottomColor": "rgba(0,0,0,0.15)",
        "position": "absolute",
        "right": "9px",
        "top": "-16px",
      }
    } />
  <div
    style={
      Object {
        "border": "7px solid transparent",
        "borderBottomColor": "#fff",
        "position": "absolute",
        "right": "10px",
        "top": "-14px",
      }
    } />
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "height": "25px",
          "width": "25px",
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "background": "#B80000",
            "cursor": "pointer",
            "height": "100%",
            "width": "100%",
          }
        }
        title="#B80000" />
    </div>
  </span>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "height": "25px",
          "width": "25px",
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "background": "#DB3E00",
            "cursor": "pointer",
            "height": "100%",
            "width": "100%",
          }
        }
        title="#DB3E00" />
    </div>
  </span>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "height": "25px",
          "width": "25px",
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "background": "#FCCB00",
            "cursor": "pointer",
            "height": "100%",
            "width": "100%",
          }
        }
        title="#FCCB00" />
    </div>
  </span>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "height": "25px",
          "width": "25px",
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "background": "#008B02",
            "cursor": "pointer",
            "height": "100%",
            "width": "100%",
          }
        }
        title="#008B02" />
    </div>
  </span>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "height": "25px",
          "width": "25px",
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "background": "#006B76",
            "cursor": "pointer",
            "height": "100%",
            "width": "100%",
          }
        }
        title="#006B76" />
    </div>
  </span>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "height": "25px",
          "width": "25px",
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "background": "#1273DE",
            "cursor": "pointer",
            "height": "100%",
            "width": "100%",
          }
        }
        title="#1273DE" />
    </div>
  </span>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "height": "25px",
          "width": "25px",
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "background": "#004DCF",
            "cursor": "pointer",
            "height": "100%",
            "width": "100%",
          }
        }
        title="#004DCF" />
    </div>
  </span>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "height": "25px",
          "width": "25px",
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "background": "#5300EB",
            "cursor": "pointer",
            "height": "100%",
            "width": "100%",
          }
        }
        title="#5300EB" />
    </div>
  </span>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "height": "25px",
          "width": "25px",
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "background": "#EB9694",
            "cursor": "pointer",
            "height": "100%",
            "width": "100%",
          }
        }
        title="#EB9694" />
    </div>
  </span>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "height": "25px",
          "width": "25px",
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "background": "#FAD0C3",
            "cursor": "pointer",
            "height": "100%",
            "width": "100%",
          }
        }
        title="#FAD0C3" />
    </div>
  </span>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "height": "25px",
          "width": "25px",
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "background": "#FEF3BD",
            "cursor": "pointer",
            "height": "100%",
            "width": "100%",
          }
        }
        title="#FEF3BD" />
    </div>
  </span>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "height": "25px",
          "width": "25px",
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "background": "#C1E1C5",
            "cursor": "pointer",
            "height": "100%",
            "width": "100%",
          }
        }
        title="#C1E1C5" />
    </div>
  </span>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "height": "25px",
          "width": "25px",
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "background": "#BEDADC",
            "cursor": "pointer",
            "height": "100%",
            "width": "100%",
          }
        }
        title="#BEDADC" />
    </div>
  </span>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "height": "25px",
          "width": "25px",
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "background": "#C4DEF6",
            "cursor": "pointer",
            "height": "100%",
            "width": "100%",
          }
        }
        title="#C4DEF6" />
    </div>
  </span>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "height": "25px",
          "width": "25px",
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "background": "#BED3F3",
            "cursor": "pointer",
            "height": "100%",
            "width": "100%",
          }
        }
        title="#BED3F3" />
    </div>
  </span>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "height": "25px",
          "width": "25px",
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "background": "#D4C4FB",
            "cursor": "pointer",
            "height": "100%",
            "width": "100%",
          }
        }
        title="#D4C4FB" />
    </div>
  </span>
</div>
`;

exports[`test Github renders correctly 1`] = `
<div
  className="github-picker"
  style={
    Object {
      "MozBorderRadius": "4px",
      "MozBoxShadow": "0 3px 12px rgba(0,0,0,0.15)",
      "OBorderRadius": "4px",
      "OBoxShadow": "0 3px 12px rgba(0,0,0,0.15)",
      "WebkitBorderRadius": "4px",
      "WebkitBoxShadow": "0 3px 12px rgba(0,0,0,0.15)",
      "background": "#fff",
      "border": "1px solid rgba(0,0,0,0.2)",
      "borderRadius": "4px",
      "boxShadow": "0 3px 12px rgba(0,0,0,0.15)",
      "display": "flex",
      "flexWrap": "wrap",
      "msBorderRadius": "4px",
      "msBoxShadow": "0 3px 12px rgba(0,0,0,0.15)",
      "padding": "5px",
      "position": "relative",
      "width": "200px",
    }
  }>
  <div
    style={
      Object {
        "border": "8px solid transparent",
        "borderBottomColor": "rgba(0,0,0,0.15)",
        "left": "9px",
        "position": "absolute",
        "top": "-16px",
      }
    } />
  <div
    style={
      Object {
        "border": "7px solid transparent",
        "borderBottomColor": "#fff",
        "left": "10px",
        "position": "absolute",
        "top": "-14px",
      }
    } />
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "height": "25px",
          "width": "25px",
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "background": "#B80000",
            "cursor": "pointer",
            "height": "100%",
            "width": "100%",
          }
        }
        title="#B80000" />
    </div>
  </span>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "height": "25px",
          "width": "25px",
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "background": "#DB3E00",
            "cursor": "pointer",
            "height": "100%",
            "width": "100%",
          }
        }
        title="#DB3E00" />
    </div>
  </span>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "height": "25px",
          "width": "25px",
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "background": "#FCCB00",
            "cursor": "pointer",
            "height": "100%",
            "width": "100%",
          }
        }
        title="#FCCB00" />
    </div>
  </span>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "height": "25px",
          "width": "25px",
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "background": "#008B02",
            "cursor": "pointer",
            "height": "100%",
            "width": "100%",
          }
        }
        title="#008B02" />
    </div>
  </span>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "height": "25px",
          "width": "25px",
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "background": "#006B76",
            "cursor": "pointer",
            "height": "100%",
            "width": "100%",
          }
        }
        title="#006B76" />
    </div>
  </span>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "height": "25px",
          "width": "25px",
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "background": "#1273DE",
            "cursor": "pointer",
            "height": "100%",
            "width": "100%",
          }
        }
        title="#1273DE" />
    </div>
  </span>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "height": "25px",
          "width": "25px",
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "background": "#004DCF",
            "cursor": "pointer",
            "height": "100%",
            "width": "100%",
          }
        }
        title="#004DCF" />
    </div>
  </span>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "height": "25px",
          "width": "25px",
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "background": "#5300EB",
            "cursor": "pointer",
            "height": "100%",
            "width": "100%",
          }
        }
        title="#5300EB" />
    </div>
  </span>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "height": "25px",
          "width": "25px",
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "background": "#EB9694",
            "cursor": "pointer",
            "height": "100%",
            "width": "100%",
          }
        }
        title="#EB9694" />
    </div>
  </span>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "height": "25px",
          "width": "25px",
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "background": "#FAD0C3",
            "cursor": "pointer",
            "height": "100%",
            "width": "100%",
          }
        }
        title="#FAD0C3" />
    </div>
  </span>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "height": "25px",
          "width": "25px",
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "background": "#FEF3BD",
            "cursor": "pointer",
            "height": "100%",
            "width": "100%",
          }
        }
        title="#FEF3BD" />
    </div>
  </span>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "height": "25px",
          "width": "25px",
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "background": "#C1E1C5",
            "cursor": "pointer",
            "height": "100%",
            "width": "100%",
          }
        }
        title="#C1E1C5" />
    </div>
  </span>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "height": "25px",
          "width": "25px",
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "background": "#BEDADC",
            "cursor": "pointer",
            "height": "100%",
            "width": "100%",
          }
        }
        title="#BEDADC" />
    </div>
  </span>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "height": "25px",
          "width": "25px",
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "background": "#C4DEF6",
            "cursor": "pointer",
            "height": "100%",
            "width": "100%",
          }
        }
        title="#C4DEF6" />
    </div>
  </span>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "height": "25px",
          "width": "25px",
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "background": "#BED3F3",
            "cursor": "pointer",
            "height": "100%",
            "width": "100%",
          }
        }
        title="#BED3F3" />
    </div>
  </span>
  <span
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <div
      style={
        Object {
          "height": "25px",
          "width": "25px",
        }
      }>
      <div
        onClick={[Function]}
        style={
          Object {
            "background": "#D4C4FB",
            "cursor": "pointer",
            "height": "100%",
            "width": "100%",
          }
        }
        title="#D4C4FB" />
    </div>
  </span>
</div>
`;

exports[`test GithubSwatch renders correctly 1`] = `
<span
  onMouseOut={[Function]}
  onMouseOver={[Function]}>
  <div
    style={
      Object {
        "height": "25px",
        "width": "25px",
      }
    }>
    <div
      onClick={[Function]}
      style={
        Object {
          "background": "#333",
          "cursor": "pointer",
          "height": "100%",
          "width": "100%",
        }
      }
      title="#333" />
  </div>
</span>
`;
